### PR TITLE
Rlp meltdown gerhard

### PR DESCRIPTION
**Pull request by kevsul**

### CODE REVIEW

File: plasma_framework/contracts/src/utils/RLPReader.sol [hunk 1]
```diff
@@ -114,23 +114,23 @@ library RLPReader {
...
-        require(itemLen <= item.len, "Decoded length is greater than input data");
+        require(itemLen == item.len, "Decoded item length must be equal to the input data length");
 
-        uint256 len = itemLen - offset;
+        uint256 dataLen = itemLen - offset;
 
         uint result;
-        uint byte0;
-        uint memPtr = item.memPtr + offset;
+        uint byte0Data;
```
Comment (kevsul) [inline plasma_framework/contracts/src/utils/RLPReader.sol]: If we change these variable names here, we should change them everywhere else as well for consistency

File: plasma_framework/contracts/src/utils/RLPReader.sol [hunk 2]
```diff
@@ -197,8 +196,9 @@ library RLPReader {
             // Item is a single byte
             decodedLength = 1;
             offset = 0;
-        } else if (byte0 < STRING_LONG_START) {
-            // Item is a short string (length <= 55 bytes)
+        } else if (STRING_SHORT_START <= byte0 && byte0 < STRING_LONG_START) {
```
Comment (kevsul) [inline plasma_framework/contracts/src/utils/RLPReader.sol]: Nitpick: the added `STRING_SHORT_START <= byte0` (and similar in the other if statements) is unnecessary and costs extra gas.
I guess you put it in to make the code cleaner, I'm in favour of that and am happy to leave it there. Just wanted to point out the extra (probably minimal) gas cost

File: plasma_framework/test/src/utils/RLP.test.js [hunk 3]
```diff
@@ -36,8 +47,11 @@ contract('RLP', () => {
         await testNumberDecoded(this.test.decodeUint, 100);
     });
 
-    it('should decode int 0', async () => {
-        await testNumberDecoded(this.test.decodeInt, 0);
+    it('should decode 0x00', async () => {
```
Comment (kevsul) [inline plasma_framework/test/src/utils/RLP.test.js]: It turns out that toUint() should _not_ decode `0x00`.
I'll change this test when I fix that

File: plasma_framework/contracts/src/utils/RLPReader.sol (hunk 2 above)
Comment (thec00n) [reply plasma_framework/contracts/src/utils/RLPReader.sol]: Agree on the extra gas cost, it just drove me mad to jump back and forth 3 times until you are sure again what range byte0 has. I am ok if you want to remove them again

File: plasma_framework/contracts/src/utils/RLPReader.sol (hunk 1 above)
Comment (thec00n) [reply plasma_framework/contracts/src/utils/RLPReader.sol]: Agree. Should I look through the code and make the changes?

**Pull request merged**